src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -621,6 +621,34 @@ public function setPublic()
     }
 
     /**
+     * Marks the response as "immutable".
+     *
+     * @param bool $immutable Enables or disabled the immutable directive.
+     *
+     * @return $this
+     */
+    public function setImmutable($immutable = true)
```

Comment:
OK I understand. My idea was calling the `setImmutable` method without argument like setPublic/setPrivate method.

```php
return (new Response('SOME RESPONSE DATA'))
  ->setPublic()
  ->setImmutable();
```

```php
return (new Response('SOME RESPONSE DATA'))
  ->setPublic()
  ->setImmutable(true);
```

I like the first example of code (calling `setImmutable` without boolean value). Maybe it will be better to implement opposite method `setMutable` (which remove the flag immutable from the `Cache-Control` header) and to remove the argument from `setImmutable` method completely. What is your opinion about it?